### PR TITLE
fix entrypoint into notification lambda for sqs and add functional test for notification

### PR DESCRIPTION
**Pull request by parthshahva**
fix entrypoint into notification lambda for sqs and add complete functional test for notification processing

### CODE REVIEW

File: matrix/common/aws/redshift_handler.py [hunk 1]
```diff
@@ -50,3 +50,12 @@ def transaction(self, queries: typing.List[str]):
             cursor.execute(query)
         conn.commit()
         conn.close()
+
+    def run_query_and_return_results(self, query):
+        conn = pg.connect(self.database_uri)
+        cursor = conn.cursor()
+        cursor.execute(query)
+        results = cursor.fetchall()
```
Comment (calvinnhieu) [inline matrix/common/aws/redshift_handler.py]: Can this be added to `transaction` as an optional param instead?

File: tests/functional/test_matrix_service.py [hunk 2]
```diff
@@ -88,8 +90,26 @@ def test_matrix_service_without_specified_output(self):
...
 
+    @unittest.skipUnless(os.getenv('DEPLOYMENT_STAGE') != "prod",
+                         "Do not want to process fake notifications in production.")
     def test_dss_notification(self):
-        self._post_notification(bundle_fqid=INPUT_BUNDLE_IDS[self.dss_env][0])
+        bundle_fqid = INPUT_BUNDLE_IDS[self.dss_env][0]
+
+        self._post_notification(bundle_fqid=bundle_fqid, event_type="DELETE")
+        WaitFor(self._poll_db_get_analysis_row_count_from_fqid, bundle_fqid)\
+            .to_return_value(0, timeout_seconds=300)
```
Comment (calvinnhieu) [inline tests/functional/test_matrix_service.py]: sugg: `DELETE` and `TOMBSTONE` are trivial updates (remove row) compared to `CREATE` (merge tables) and could use a shorter timeout (60s)

Comment (parthshahva) [reply tests/functional/test_matrix_service.py]: Sounds good

File: matrix/common/aws/redshift_handler.py (hunk 1 above)
Comment (parthshahva) [reply matrix/common/aws/redshift_handler.py]: Sounds good

**Pull request merged**